sql/core/src/main/scala/org/apache/spark/sql/KeyValueGroupedDataset.scala

```diff
@@ -219,6 +219,160 @@ class KeyValueGroupedDataset[K, V] private[sql](
   }
 
   /**
+   * ::Experimental::
+   * (Scala-specific)
+   * Applies the given function to each group of data, while using an additional keyed state.
+   * For each unique group, the function will be passed the group key and an iterator that contains
+   * all of the elements in the group. The function can return an object of arbitrary type, and
+   * optionally update or remove the corresponding state. The returned object will form a new
+   * [[Dataset]].
+   *
+   * This function can be applied on both batch and streaming Datasets. With a streaming dataset,
+   * this function will be once for each in every trigger. For each key, the updated state from the
+   * function call in a trigger will be the state available in the function call in the next
```

Comment:
Any updates to the state will be stored and passed to the user given function in subsequent batches when executed as a Streaming Query.